src/github/github.service.ts

```diff
@@ -0,0 +1,180 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import {
+  GITHUB_CLIENT_ID,
+  GITHUB_PRIVATE_KEY,
+  REDIS_PUBSUB,
+  VAULT_KV_APPS_MOUNT,
+} from '../constants';
+import axios, { AxiosInstance } from 'axios';
+import * as jwt from 'jsonwebtoken';
+import sodium from 'libsodium-wrappers';
+import { RedisService } from '../redis/redis.service';
+import { VaultService } from '../vault/vault.service';
+
+@Injectable()
+export class GithubService implements OnModuleInit {
+  private readonly axiosInstance: AxiosInstance;
+  private readonly clientId = GITHUB_CLIENT_ID;
+  private readonly privateKey = GITHUB_PRIVATE_KEY;
+
+  constructor(
+    private readonly vaultService: VaultService,
+    private readonly redisService: RedisService,
+  ) {
+    this.axiosInstance = axios.create({
+      baseURL: 'https://api.github.com',
+      headers: {
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+  }
+
+  onModuleInit() {
+    // Subscribe to the Redis channel
+    this.redisService.subscribeAndProcess(REDIS_PUBSUB.VAULT_SERVICE_TOKEN, async (event) => {
+      try {
+        const { clientId, environment, project, service, scmUrl } = event.data as {
+          clientId: string;
+          environment: string;
+          project: string;
+          service: string;
+          scmUrl: string;
+        };
+        const path = `${environment}/${project}/${service}`;
+        const kvData= await this.vaultService.getKv(VAULT_KV_APPS_MOUNT,path);
+        const value = kvData[`broker-jwt:${clientId}`];
+        if (!value) {
+          console.error(`No value found for client_id: ${clientId}`);
+          return;
+        }
+        if (scmUrl) {
```

Comment:
Check if scmUrl is GitHub. Bonus points if it checks the org and if the GitHub app has access.